src/test/java/org/fcrepo/client/ConnectionManagementIT.java

```diff
@@ -0,0 +1,371 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.client;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.input.NullInputStream;
+import org.apache.commons.io.output.NullOutputStream;
+import org.apache.http.HttpClientConnection;
+import org.apache.http.HttpStatus;
+import org.apache.http.conn.HttpClientConnectionManager;
+import org.apache.http.conn.routing.HttpRoute;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.mockserver.client.server.MockServerClient;
+import org.mockserver.junit.MockServerRule;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import static org.fcrepo.client.MockHttpExpectations.host;
+import static org.fcrepo.client.MockHttpExpectations.port;
+import static org.fcrepo.client.TestUtils.TEXT_TURTLE;
+import static org.fcrepo.client.TestUtils.setField;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Integration test used to demonstrate connection management issues with the FcrepoClient.
+ *
+ * @author esm
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class ConnectionManagementIT {
+
+    /**
+     * Starts a mock HTTP server on a free port
+     */
+    @Rule
+    public MockServerRule mockServerRule = new MockServerRule(this);
+
+    // Set by the above @Rule, initialized on @Before via MockHttpExpectations
+    private MockServerClient mockServerClient;
+
+    /**
+     * URIs that our Mock HTTP server responds to.
+     */
+    private MockHttpExpectations.SupportedUris uris;
+
+    /**
+     * Verifications for HttpClientConnectionManager
+     */
+    static class Verifications {
+
+        /**
+         * Verifies that the expected number of connections have been requested and then closed.
+         *
+         * @param connectionCount the number of connections that have been requested and closed.
+         * @return a Consumer that verifies the supplied HttpClientConnectionManager has opened and closed the expected
+         * number of connections.
+         */
+        static Consumer<HttpClientConnectionManager> connectionRequestedAndClosed(final int connectionCount) {
```

Comment:
It's not totally clear to me where the value of making this a `Consumer` (and not just a `static void` method with a `HttpClientConnectionManager` param) is coming in?